### PR TITLE
Add org.kde.kalk

### PR DESCRIPTION
**Pull request by espidev**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://invent.kde.org/plasma-mobile/kalk (I am a maintainer)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
A backport patch was needed for appdata, it will be in the next release.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by espidev** [comment]
bot, build org.kde.kalk

**Comment by barthalion** [comment]
Fails appdata validation:

```
builddir/files/share/appdata/org.kde.kalk.appdata.xml: FAILED:
• style-invalid         : Not enough <screenshot> tags, minimum is 1
```

**Comment by espidev** [comment]
bot, build org.kde.kalk

File: appdata-fix.patch [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
++  <project_license>GPL-3.0-or-later</project_license>
++  <content_rating type="oars-1.1"/>
+   <name>Calculator</name>
+   <name xml:lang="ca">Calculadora</name>
+   <name xml:lang="ca-valencia">Calculadora</name>
+@@ -77,29 +80,30 @@
+   <developer_name xml:lang="x-test">xxKDE Communityxx</developer_name>
+   <metadata_license>CC0-1.0</metadata_license>
+   <description>
+-    <p>Kalk is a powerful calculator designed for mobile platform.</p>
```
Comment (ognarb) [inline appdata-fix.patch:19]: Not sure why you changed the description.

File: appdata-fix.patch [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+-    <p xml:lang="x-test">xxKalk is a powerful calculator designed for mobile platform.xx</p>
++    <p>Kalk is a convergent calculator for Plasma.</p>
+   </description>
++  <screenshots>
++    <screenshot>
++      <caption>Calculator Page</caption>
++      <image>https://cdn.kde.org/screenshots/kalk/kalk.png</image>
++    </screenshot>
++  </screenshots>
++  <recommends>
```
Comment (ognarb) [inline appdata-fix.patch:51]: We should put this on all of our apps :( I will take care of it upstream. @barthalion Should I try to add this change as patches to the maui/plasma mobile Flathub MR I made too? Or it is fine if they appear only with the 21.06 release we will publish next month?

File: appdata-fix.patch (hunk 1 above)
Comment (espidev) [reply appdata-fix.patch:19]: I had an MR merged upstream for this, but basically the old description is grammatically incorrect and didn't convey that the application works on desktop as well.

Comment (ognarb) [reply appdata-fix.patch:19]: ok :)

File: appdata-fix.patch (hunk 2 above)
Comment (espidev) [reply appdata-fix.patch:51]: There was another tag of interest, `display_length` that I wasn't sure if we should start adopting for stating that it works on mobile screen sizes.

https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html

Comment (barthalion) [reply appdata-fix.patch:51]: It's fine to add things later.

**Comment by espidev** [comment]
bot, build org.kde.kalk

**Comment by espidev** [comment]
bot, build org.kde.kalk

**Pull request closed without merging**